README.md

```diff
@@ -186,3 +186,161 @@ If none package suits scope of new changes, we need to create new package by cre
+);
+
+export default AlertWrapper;
+
+
+```
+
+3. Always export nested types. It can be required by a consumer app when constructing component props.
+
+```TS
+import React from 'react';
+import { Alert, AlertProps, AlertVariant } from '@patternfly/react-core';
+
+export type CustomType = 'A' | 'B' | 2 | 'something'
+export interface CustomInterface {
+  a: string,
+  b?: number
+}
+
+export interface AlertWrapperProps extends Omit<AlertProps, 'variant'> {
+  nestedNonPrimitiveType: CustomType
+  nestedNonPrimitiveInterface: CustomInterface
+  nestedNonPrimitiveArray: CustomType[]
+}
+
+const AlertWrapper: React.FunctionComponent<AlertWrapperProps> = ({ specificRequiredProp, ...props }) => (
+    <div>
+        <span>{specificRequiredProp}</span>
+        <Alert {...props} variant={AlertVariant.danger} />
+    </div>
+);
+
+export default AlertWrapper;
+
+```
+
+### General guidelines
+
+#### Names
+
+1. Use PascalCase for type names.
+2. Do not use I as a prefix for interface names.
+3. Use PascalCase for enum values.
+4. Use camelCase for function names.
+5. Use camelCase for property names and local variables.
+6. Do not use _ as a prefix for private properties.
+7. Use whole words in names when possible.
+
+#### Comments
+
+1. Always add a short description to exported types. The description is used for generating documentation
+2. Use JSDoc style comments for functions, interfaces, enums, and classes.
+
+```TS
+export interface AlertWrapperProps extends Omit<AlertProps, 'variant'> {
+  /**
+   * Alert id
+  */
+  id: string,
+  /**
+   * A content displayed as Patternfly Alert children
+  */
+  description: React.ReactNode
+}
+
+```
+
+#### Types
+
+1. Do not export types/functions unless you need to share it across multiple components.
+2. Do not introduce new types/values to the global namespace.
+3. Shared types should be defined in types.ts.
+4. Within a file, type definitions should come first.
+
+#### null and undefined
+
+1. Use undefined. Do not use null.
+
+#### General Constructs
+
+For a variety of reasons, we avoid certain constructs, and use some of our own. Among them:
+
+1. Do not use for..in statements; instead, use ts.forEach, ts.forEachKey and ts.forEachValue. Be aware of their slightly different semantics.
+2. Try to use ts.forEach, ts.map, and ts.filter instead of loops when it is not strongly inconvenient.
+
+#### Style
+
+1. Use arrow functions over anonymous function expressions.
+2. Open curly braces always go on the same line as whatever necessitates them.
+3. Parenthesized constructs should have no surrounding whitespace.
+4. A single space follows commas, colons, and semicolons in those constructs. For example:
+```ts
+for (var i = 0, n = str.length; i < 10; i++) { }
+if (x < 10) { }
+function f(x: number, y: string): void { }
+```
+5. Use a single declaration per variable statement
+```ts
+(i.e. use var x = 1; var y = 2; over var x = 1, y = 2;).
+```
+6. else goes on a separate line from the closing curly brace.
```

Comment:
I would remove this rule, can we also add to use style rule about single quotes?

Reply:
I will remove this one. For quotes, let's handle that through eslint.